lib/cloudregister/registerutils.py

```diff
@@ -531,31 +531,35 @@ def setup_registry(registry_fqdn, username, password):
 
 
 # ----------------------------------------------------------------------------
-def get_registry_credentials():
+def get_registry_credentials(set_new):
     """Read the registry credentials file
        and return its content or an empty dict."""
     config_json = {}
-    failed = False
-    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
-        try:
-            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
-                config_json = json.load(cred_json)
-        except json.decoder.JSONDecodeError:
-            logging.info(
-                'Unable to parse existing %s, preserving file as %s.bak, '
-                'writing new credentials' % (
-                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-                )
-            )
-            # we want to remove the possible
-            # auth token present in the file
-            # before preserving it
-            mv_file_cmd = 'mv -Z {} {}.bak'.format(
-                REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-            ).split()
-            failed = exec_subprocess(mv_file_cmd)
-            message = 'File not preserved.' if failed else 'File preserved.'
-            logging.info(message)
+    failed = None
+    if not os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        logging.info('Credentials file does not exist. Nothing to do')
+        return config_json, failed
+
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+            config_json = json.load(cred_json)
+    except json.decoder.JSONDecodeError:
```

Comment:
This is not catching opening errors, e.g permission denied. Intentional ?

Reply:
Yes, it will get the [code](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L543) from #159 once merged into this branch

Reply:
master branch merged and conflicts fixed, tests updated